src/System.Management.Automation/engine/remoting/commands/InvokeCommandCommand.cs

```diff
@@ -1162,7 +1197,19 @@ protected override void ProcessRecord()
                 }
                 else if (ParameterSetName.Equals(InvokeCommandCommand.InProcParameterSet) && (_steppablePipeline != null))
                 {
-                    _steppablePipeline.Process(InputObject);
+                    try
+                    {
+                        _steppablePipeline.Process(InputObject);
+                    }
+                    catch
+                    {
+                        if (MyInvocation.BoundParameters.ContainsKey(nameof(StrictModeVersion)))
+                        {
+                            Context.EngineSessionState.CurrentScope.StrictModeVersion = _savedStrictModeVersion;
+                        }
```

Comment:
It's better to put this in a finally block instead, e.g.

```
    try
    {
        _steppablePipeline.Process(InputObject);
    }
    finally
    {
        if (_savedStrictModeVersion is not null)
        {
            Context.EngineSessionState.CurrentScope.StrictModeVersion = _savedStrictModeVersion;
        }
    }
```

Reply:
Context.EngineSessionState.CurrentScope.StrictModeVersion is only reverted at End block or on error.  Otherwise strict mode is incorrect for other Processing block streaming input.

Reply:
Oh, that makes sense.